tests/many_to_many/tests.py

```diff
@@ -554,3 +555,13 @@ def test_inherited_models_selects(self):
             ]
         )
         self.assertQuerysetEqual(b.publications.all(), ['<Publication: Science Weekly>'])
+
+    def test_count_query(self):
+        """
+        #29725 - Calling count() on a many to many relation should not generate inefficient sql
+        """
+        article = self.a1
+
+        with CaptureQueriesContext(connection) as captured_query:
+            article.publications.count()
+            self.assertNotIn('INNER JOIN', captured_query[0]['sql'])
```

Comment:
Might want to only test for `JOIN` presence as this wouldn't fail if `LEFT JOIN` was used.